test/gtest/common/mem_buffer.cc

```diff
@@ -153,31 +153,54 @@ size_t mem_buffer::get_bar1_free_size()
 {
     /* All gtest CUDA tests explicitly assume that all memory allocations are
      * done on the device 0. The same assumption is followed here. */
-    size_t availible_size = SIZE_MAX;
+    static size_t available_size = SIZE_MAX;
 
 #if HAVE_NVML_H
     nvmlReturn_t ret;
     nvmlDevice_t device;
     nvmlBAR1Memory_t bar1mem;
 
+    /* No need to call nvml if the size was trimmed earlier */
+    if (available_size != SIZE_MAX) {
+        return available_size;
+    }
+
+    ret = nvmlInit_v2();
+    if (ret != NVML_SUCCESS) {
+        UCS_TEST_MESSAGE << "nvmlInit_v2 failed: " << nvmlErrorString(ret) <<
+                            ", error code: " << ret;
+        return available_size;
+    }
+
     ret = nvmlDeviceGetHandleByIndex(0, &device);
     if (ret != NVML_SUCCESS) {
         /* For whatever reason we cannot open device handle.
          * As a result let's assume there is no limit on the size
          * and in the worse case scenario gtest will fail in runtime */
-        return availible_size;
+        UCS_TEST_MESSAGE << "nvmlDeviceGetHandleByIndex failed: " <<
+        nvmlErrorString(ret) << ", error code: " << ret;
```

Comment:
add indent
see https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=56897&view=logs&j=cc064a77-22b5-56bf-ecc0-70b5fe764261&t=aedfd754-44a6-53e6-6843-8659d800fee2 for code style issues